Xamarin.Forms.Core/IndicatorsView.cs

```diff
@@ -0,0 +1,285 @@
+		{
+			(bindable as IndicatorsView)?.ResetIndicatorsStyles();
+		});
+
+		public static readonly BindableProperty IndicatorsColorProperty = BindableProperty.Create(nameof(IndicatorsColor), typeof(Color), typeof(IndicatorsView), Color.Silver, propertyChanged: (bindable, oldValue, newValue) =>
+		{
+			(bindable as IndicatorsView)?.ResetIndicatorsStyles();
+		});
+
+		public static readonly BindableProperty SelectedIndicatorColorProperty = BindableProperty.Create(nameof(SelectedIndicatorColor), typeof(Color), typeof(IndicatorsView), Color.Gray, propertyChanged: (bindable, oldValue, newValue) =>
+		{
+			(bindable as IndicatorsView)?.ResetIndicatorsStyles();
+		});
+
+		public static readonly BindableProperty IndicatorsSizeProperty = BindableProperty.Create(nameof(IndicatorsSize), typeof(double), typeof(IndicatorsView), 10.0, propertyChanged: (bindable, oldValue, newValue) =>
+		{
+			(bindable as IndicatorsView)?.ResetIndicatorsStyles();
+		});
+
+		internal static readonly BindableProperty ItemsSourceProperty = BindableProperty.Create(nameof(ItemsSource), typeof(IEnumerable), typeof(IndicatorsView), null, propertyChanged: (bindable, oldValue, newValue) =>
+		{
+			(bindable as IndicatorsView)?.ResetItemsSource(oldValue as IEnumerable);
+		});
+
+		static IndicatorsView()
+		{
+		}
+
+		public IndicatorsView()
+		{
+			Orientation = StackOrientation.Horizontal;
+		}
+
+		public int Position
+		{
+			get => (int)GetValue(PositionProperty);
+			set => SetValue(PositionProperty, value);
+		}
+
+		public int Count
+		{
+			get => (int)GetValue(CountProperty);
+			set => SetValue(CountProperty, value);
+		}
+
+		public int MaxVisibleCount
+		{
+			get => (int)GetValue(MaxVisibleCountProperty);
+			set => SetValue(MaxVisibleCountProperty, value);
+		}
+
+		public DataTemplate IndicatorTemplate
+		{
+			get => GetValue(IndicatorTemplateProperty) as DataTemplate;
+			set => SetValue(IndicatorTemplateProperty, value);
+		}
+
+		public IndicatorShape IndicatorShape
+		{
+			get => (IndicatorShape)GetValue(IndicatorShapeProperty);
+			set => SetValue(IndicatorShapeProperty, value);
+		}
+
+		public Color IndicatorsColor
+		{
+			get => (Color)GetValue(IndicatorsColorProperty);
+			set => SetValue(IndicatorsColorProperty, value);
+		}
+
+		public Color SelectedIndicatorColor
+		{
+			get => (Color)GetValue(SelectedIndicatorColorProperty);
+			set => SetValue(SelectedIndicatorColorProperty, value);
+		}
+
+		public double IndicatorsSize
+		{
+			get => (double)GetValue(IndicatorsSizeProperty);
+			set => SetValue(IndicatorsSizeProperty, value);
+		}
+
+		[EditorBrowsable(EditorBrowsableState.Never)]
+		public IEnumerable ItemsSource
+		{
+			get => GetValue(ItemsSourceProperty) as IEnumerable;
+			set => SetValue(ItemsSourceProperty, value);
+		}
+
+		[EditorBrowsable(EditorBrowsableState.Never)]
+		public new IList<View> Children => base.Children;
+
+		public static void SetItemsSourceBy(IndicatorsView indicatorsView, CarouselView carouselView)
```

Comment:
naming is very confusing. and although I understand the use of such an helper method, I don't think it should be part of the class. maybe move it to an extensions class instead ?